br/pkg/restore/snap_client/client.go

```diff
@@ -672,12 +672,43 @@ func (rc *SnapClient) getRebasedTables() map[restore.UniqueTableName]bool {
 	return rc.rebasedTablesMap
 }
 
+// CreateTables create tables, and generate their information.
+// this function will use workers as the same number of sessionPool,
+// leave sessionPool nil to send DDLs sequential.
+func (rc *SnapClient) CreateTables(
+	ctx context.Context,
+	tables []*metautil.Table,
+	newTS uint64,
+) ([]*CreatedTable, error) {
+	log.Info("start create tables", zap.Int("total count", len(tables)))
+	rc.generateRebasedTables(tables)
+
+	// try to restore tables in batch
+	if rc.batchDdlSize > minBatchDdlSize && len(rc.dbPool) > 0 {
+		tables, err := rc.createTablesBatch(ctx, tables, newTS)
+		if err == nil {
+			return tables, nil
+		} else if !utils.FallBack2CreateTable(err) {
+			return nil, errors.Trace(err)
+		}
+		// fall back to old create table (sequential create table)
+		log.Info("fall back to the sequential create table")
+	}
+
+	// restore tables in db pool
+	if len(rc.dbPool) > 0 {
+		return rc.createTablesSingle(ctx, rc.dbPool, tables, newTS)
+	}
+	// restore tables in one db
```

Comment:
Can this `non-dbPool` part be removed?

Reply:
we can unify the `rc.dbPool` and `rc.db` by another PR.